source/common/upstream/cds_api_impl.cc

```diff
@@ -119,12 +122,14 @@ void CdsApiImpl::onConfigUpdate(
     }
   }
 
+  if (any_applied) {
+    system_version_info_ = system_version_info;
```

Comment:
Is this an unrelated change? Should it be split out or is it related enough to include?

Reply:
Yeah, this could reasonably be split out. This is a bug I came across while doing this work.